shared/logging/logger.go

```diff
@@ -0,0 +1,37 @@
+package logging
+
+import (
+	"log/slog"
+	"os"
+)
+
+// Level is the current log level of Default. To change the level at runtime, for example to DEBUG, call Level.Set(slog.LevelDebug)
+// Defaults to slog.LevelInfo
+var Level = new(slog.LevelVar)
+
+// Default is a *slog.Logger configured with a JSON handler and a level set by environment variable LOG_LEVEL
+// If LOG_LEVEL is not set, or is set to an unknown value, level defaults to slog.LevelInfo
+var Default *slog.Logger
+
+func init() {
+	configureLogging()
+}
+
+// configureLogging separated out from init() for testing with environment variables
+func configureLogging() {
+	envLogLevel := os.Getenv("LOG_LEVEL")
+	if len(envLogLevel) > 0 {
```

Comment:
Should we capture an `else` case for this `if` block?

Reply:
The `else` clause would just be to use the default level of `INFO`. But because `slog` helpfully makes the zero value of `slog.LevelVar` `INFO` already there is nothing to do.

But I think I will rework this to distinguish `LOG_LEVEL` not being set (a normal turn of events that results in an `INFO` level) and `LOG_LEVEL` being set, but the value being empty (still sets the level to `INFO` but logs a warning about an empty `LOG_LEVEL` value since that may be an unintentional oversight or typo.)